### ISSUE TITLE
Page template issues after upgrading to flux 9.3.2

### ISSUE THREAD

**Issue by mkrappitz** [FluidTYPO3/flux#1742]
We are having a TYPO3 9 installation with flux 9.2 for content elements and backend layout (In Page TS:  mod.web_layout.BackendLayouts) based page templating with fluid templates (page.10 = FLUIDTEMPLATE) in an provider extension EXT:template. No fluidpages, all working fine. After upgrading flux 9.2 to 9.3.2 several to my mind weird stuff happens:

1. In frontend after the opening <body> tag there is a message printed: "Please select a page template in page properties." We have working backend layout based page templating and don't want anything else. Why this message? Are we now forced to use fluidpage based page templating?
2. In the root of our TYPO3 installation a folder named "design" got created with substructure Layouts, Partials, Templates. We don't want and need this folder, we have a provider extension with these folders in EXT:template/Resources/Private/.
3. To try to get rid of the described message in "1." we selected in the page properties of a page under tab "Appearance" > "Backend Layout (this page only)" and "Backend Layout (subpages of this page)" from our own backend layouts to both the flux layouts "Columns from selected Page Layout". This results in frontend in what is described in #1741 "Could not ultimately dispatch the request after 101 iterations" and in backend we have no colums in page view anymore and existing content elements are in column "Unused".
4. Also this happens after the extension upgrade for the sake of completeness of this ticket: #1740 

I don't get the intention of this release. It seems pretty broken to me, introduces several breaking changes without further notice or documentation (not counting commit messages as documentation), it seems to all of a sudden to force usage of fluidpages page templating and creates a folder structure for templating by default in web-root without asking or informing about it. I also couldn't find any option to disable the fluidpage feature in flux in general yet.
All of this I would consider as unexpected and undesirable behaviour.

Maybe I'm getting all this terribly wrong, but without documentation this is hard to tell.

Reverting to flux 9.2 for now.

**Comment by NamelessCoder** [comment]
> Are we now forced to use fluidpage based page templating?

No, but you do need to unset the default TS object that gets generated, which is `page.5`. Clear this object in your TS template and you should be good to go.

> In the root of our TYPO3 installation a folder named "design" got created

You can disable creation of this folder in Flux extension settings, but it will however be generated once and needs to be removed after disabling the "plugAndPlay" setting. The only way to completely prevent this folder from being generated is to ship extension configuration as part of the deployed site, so the setting is toggled off from the very first page load.

> Could not ultimately dispatch the request after 101 iterations

Please follow #1741, there is no known solution for this - yet. But it is high priority since I take it more than one user is now affected by it.

**Comment by mkrappitz** [comment]
Thanks, this helps already! 
Still a huge amount of flux users will run into these troubles and especially those with less technical understanding will be frustrated, also as the only source of information about it is this github ticket right now. Anyway many thanks for all your drive towards innovation in the TYPO3 project!

**Comment by NamelessCoder** [comment]
You're most welcome, @mkrappitz - I will pin this issue so others who run into the same will find the solutions more easily.

**Comment by UNI49** [comment]
Hi,

I've a similar problem, I've already override the `page.5`, and it's totally okay on front. But still the problem for template on backend, page edition "Tried resolving a template file for controller action "Page->interne" in format ".html", but none of the paths contained the expected template file ()."

Is there something to do ? Another configuration to override ?

I use Typo3 v9.5.14, Flux v9.3.2 and SItePackage for templating.

Thanks for help.

**Comment by NamelessCoder** [comment]
@UNI49 It looks like you have no template paths defined. If you have a multisite setup make sure your TS is loaded in the main/first root TS template.

**Comment by NamelessCoder** [comment]
Closing the issue since the linked problems are solved and the questions answered, except for one detail:

For users who don't wish to use the page templating feature, in addition to unsetting the `page.5` index to disable the frontend rendering you can also disable the following TCA fields:

* `pages.tx_fed_page_flexform`
* `pages.tx_fed_page_flexform_sub`
* `pages.tx_fed_page_controller_action`
* `pages.tx_fed_page_controller_action_sub`

This can be done through the usual means of manipulating TCA, either through an extension with a TCA override for `pages`, or through pageTSconfig to remove the fields from `showitem` of the `pages` table.

> I don't get the intention of this release.

There are two main reasons for including the new features:

1. The more frequent use case of Flux is to combine it with Fluidpages and it has been a bit of a pain to ensure that versions of Flux and Fluidpages are in sync. Since it's the more frequent use case the selected solution was to include the page templating feature in Flux.
2. The global design folder is added to provide an alternative method to demanding a provider extension - in order to lower the entry barrrier for people who "just want to get started" without first learning how to write a TYPO3 extension.

Since these two features are specifically aimed at novice users they are enabled by default but made possible to deactivate for more experienced users.

TL;DR: the intention of the 9.3 release is to make it easier for new users to pick up Fluid templating in TYPO3 by using Flux to handle the integration aspects.

**Comment by mkrappitz** [comment]
👍
